### PR TITLE
Data Cleaning Processors

### PR DESCRIPTION
**Pull request by ggmarshall**
I have added two processors for data cleaning. The first is a processor to invert the polarity for identifying discharge type events and the second is a time over threshold processor for looking at saturated events.

### CODE REVIEW

**Comment by gipert** [comment]
Nice. Can you please also briefly document the processors in the docstrings?

**Comment by ggmarshall** [comment]
done

File: src/pygama/dsp/processors/invert_polarity.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+@guvectorize(
+    ["void(float32[:], float32[:])", "void(float64[:], float64[:])"],
+    "(n)->(n)",
+    **nb_kwargs,
+)
+def invert_polarity(w_in: np.ndarray, w_out: np.ndarray) -> None:
+    """Inverts the polarity of the waveform.
+    Parameters
```
Comment (gipert) [inline src/pygama/dsp/processors/invert_polarity.py:16]:
```suggestion
    """Inverts the polarity of the waveform.
    
    Parameters
```
The build will fail if you don't have spaces between sections

**Comment by jasondet** [comment]
@ggmarshall is invert_polarity really necessary? Can't you get the same behavior using numpy.multiply with -1?

**Comment by jasondet** [comment]
@gipert just wondering -- how does codecov report an increase in coverage in a PR with two new files and no new tests?

**Comment by ggmarshall** [comment]
Good point, I have removed it

**Comment by gipert** [comment]
> how does codecov report an increase in coverage in a PR with two new files and no new tests?

he saw your comment and changed his mind...

**Comment by mmatteo** [comment]
Should the processor also recalibrate the adc scale such to have only positive values? Are all our processors able to handle negative adc values?

**Pull request merged**